Add Prefix character validation

Tiomat85 opened the pull request:
Fixes #1490 

When editing the prefix, add validation and if invalid disable the ability to start an export with a reason given as a tooltip.  Further development could include visual identification of the problematic section.

Tiomat85 committed:
Add Prefix character validation

When editing the prefix, add validation and if invalid disable the ability to start an export with a reason given as a tooltip.  Further development could include visual identification of the problematic section.

cmeyer commented on nion/swift/ExportDialog.py:
```diff
@@ -101,6 +101,28 @@ def on_writer_changed(self, widget: Declarative.UIWidget, current_index: int) ->
+        invalid_reason: str | None = None
+
+        prefix_text = self.viewmodel.prefix.value or ""
+        if prefix_text != Utility.simplify_filename(prefix_text) or "\\" in prefix_text or "/" in prefix_text:
```
Doesn't `simplify_filename` already catch forward and backslashes? If not, why doesn't it?

cmeyer replied:
Ping

Tiomat85 committed:
Fix race condition for binding vs event, so checking with correct value. Removed unneeded additonal checks.

cmeyer commented:
In addition to some stylistic items, I'm delaying this until the threaded export dialog is merged, which is a larger change and conflicts with this one.

cmeyer commented on nion/swift/ExportDialog.py:
```diff
@@ -101,6 +101,26 @@ def on_writer_changed(self, widget: Declarative.UIWidget, current_index: int) ->
+    def on_prefix_changed(self, widget: Declarative.UIWidget, text: str) -> None:
+        self.validate_options(text)
+
+    def validate_options(self, prefix_text: str) -> bool:
```
Can this just be merged into `_handle_prefix_changed`?

cmeyer commented on nion/swift/ExportDialog.py:
```diff
@@ -101,6 +101,26 @@ def on_writer_changed(self, widget: Declarative.UIWidget, current_index: int) ->
         writer = self.__writers[current_index]
         self.viewmodel.writer.value = writer
 
+    def on_prefix_changed(self, widget: Declarative.UIWidget, text: str) -> None:
```
Can this be named `_handle_prefix_changed`. The variables named `on_xyz` are usually function pointers, not handlers. And the `_` prefix will denote it as "as private as can be when passing to Declarative UI". Thanks.

cmeyer reviewed (changes requested):
Minor change requests.

Tiomat85 committed:
Rename on_prefix_changed to _handle_prefix_changed

Rename on_prefix_changed to _handle_prefix_changed

In addition consolidated functions together.

cmeyer commented:
The `_handle_prefix_changed` should not have a `text` parameter:

```
TypeError: ExportDialog._handle_prefix_changed() got an unexpected keyword argument 'text'
```

cmeyer commented:
Merged b7a9556a2b8608f1442179b28a241dad4f409349
